File: Python/Stack/M-2390. Removing Stars From a String.py
class Solution:
    @classmethod
    def removeStars(self, s: str) -> str:
        stack = []
        for char in s:
            if len(stack) and char == "*":
                stack.pop()
            else:
                stack.append(char)
        return "".join(stack)


testData = [
    {
        "s": "leet**cod*e",
        "ans": "lecoe",
    },
    {
        "s": "erase*****",
        "ans": "",
    },
]

for data in testData:
    print(data["ans"] == Solution.removeStars(data["s"]))

File: Stack/M-2390. Removing Stars From a String/M-2390. Removing Stars From a String.ts
function removeStars(s: string): string {
  const stack: string[] = [];
  for (const char of s) {
    if (char === '*' && stack.length) stack.pop();
    else stack.push(char);
  }
  //console.log('stack', stack);
  return stack.join('');
}

const testData = [
  {
    s: 'leet**cod*e',
    ans: 'lecoe',
  },
  {
    s: 'erase*****',
    ans: '',
  },
];

for (const data of testData) {
  console.log(data['ans'] === removeStars(data['s']));
}
